lib/variable/include/scipp/variable/accumulate.h

```diff
@@ -19,7 +20,8 @@ static void do_accumulate(const std::tuple<Ts...> &types, Op op,
   // - `other` are small, to avoid overhead (important for groupby), limit set
   //   by tuning BM_groupby_large_table
   // - reduction to scalar with more than 1 `other`
-  constexpr scipp::index small_input = 16384;
+  const bool binned_input = (is_bins(other) || ...);
+  const scipp::index small_input = binned_input ? 2 : 16384;
```

Comment:
If I understand correctly, this would pessimise the case of many (<16384) tiny bins. But I don't see a universal solution and that case should be relatively fast anyway.